storage-provider/server/src/p2p/register.rs

```diff
@@ -59,83 +61,98 @@ impl RegisterConfig {
 }
 
 /// Register the peer with the rendezvous point.
 /// The ttl is how long the peer will remain registered in seconds.
 pub(crate) async fn register(
-    mut swarm: Swarm<RegisterBehaviour>,
+    swarm: &mut Swarm<RegisterBehaviour>,
     rendezvous_point: PeerId,
     rendezvous_point_address: Multiaddr,
-    ttl: Option<u64>,
+    ttl: u64,
     namespace: Namespace,
 ) -> Result<(), P2PError> {
     tracing::info!("Attempting to register with rendezvous point {rendezvous_point} at {rendezvous_point_address}");
+    let mut register_tick = tokio::time::interval(Duration::from_secs(ttl));
+
+    // Dial into bootstrap address
     swarm.dial(rendezvous_point_address.clone())?;
+    // Get and add external address
+    let external_addr = get_external_address(swarm).await;
+    swarm.add_external_address(external_addr);
 
-    while let Some(event) = swarm.next().await {
-        match event {
-            SwarmEvent::NewListenAddr { address, .. } => {
-                tracing::info!("Listening on {}", address);
-            }
-            SwarmEvent::ConnectionClosed {
-                peer_id,
-                cause: Some(error),
-                ..
-            } if peer_id == rendezvous_point => {
-                tracing::info!("Lost connection to rendezvous point {}", error);
-            }
-            // once `/identify` did its job, we know our external address and can register
-            SwarmEvent::Behaviour(RegisterBehaviourEvent::Identify(
-                identify::Event::Received { info, .. },
-            )) => {
-                // Register our external address.
-                tracing::info!("Registering external address {}", info.observed_addr);
-                swarm.add_external_address(info.observed_addr);
-                if let Err(error) = swarm.behaviour_mut().rendezvous.register(
-                    namespace.clone(),
-                    rendezvous_point,
-                    ttl,
-                ) {
+    loop {
+        tokio::select! {
+            // Poll tick every TTL to re-register.
+            // First tick completes immediately.
+            _ = register_tick.tick() => {
+                tracing::info!("Registering with p2p node");
+                // Dial to establish a connection.
+                // Dial is needed because the connection is not kept alive using rendezvous protocol.
+                swarm.dial(rendezvous_point_address.clone())?;
+                // Register with bootstrap node.
+                if let Err(error) =
+                    swarm
+                        .behaviour_mut()
+                        .rendezvous
+                        .register(namespace.clone(), rendezvous_point, Some(ttl))
+                {
                     tracing::error!("Failed to register: {error}");
                     return Err(P2PError::RegistrationFailed(rendezvous_point));
+                } else {
+                    tracing::info!("Registration with {rendezvous_point} successful");
```

Comment:
I think that the registration is not yet confirmed on this line.
```suggestion
                    tracing::info!("Registration requested with {rendezvous_point}");
```

Reply:
Yep, you are right, it has only been requested at this point.